test/PassPlanFilePluginTest.groovy

```diff
@@ -45,35 +45,73 @@ class PassPlanFilePluginTest {
+                PassPlanFilePlugin.withDirectory('./terraform/')
+                PassPlanFilePlugin plugin = new PassPlanFilePlugin()
+                TerraformPlanCommand command = new TerraformPlanCommand("dev")
+                plugin.apply(command)
+
+                String result = command.toString()
+                assertThat(result, containsString("-out=tfplan-dev"))
+            }
+
+            @Test
+            void addsArgumentToTerraformApply() {
+                PassPlanFilePlugin.withDirectory('./terraform/')
+                PassPlanFilePlugin plugin = new PassPlanFilePlugin()
+                TerraformApplyCommand command = new TerraformApplyCommand("dev")
+                plugin.apply(command)
+
+                String result = command.toString()
+                assertThat(result, containsString("tfplan-dev"))
+            }
         }
 
-        @Test
-        void addsArgumentToTerraformPlan() {
-            PassPlanFilePlugin plugin = new PassPlanFilePlugin()
-            TerraformPlanCommand command = new TerraformPlanCommand("dev")
-            plugin.apply(command)
-
-            String result = command.toString()
-            assertThat(result, containsString("-out=tfplan-dev"))
-        }
-
-        @Test
-        void addsArgumentToTerraformApply() {
-            PassPlanFilePlugin plugin = new PassPlanFilePlugin()
-            TerraformApplyCommand command = new TerraformApplyCommand("dev")
-            plugin.apply(command)
-
-            String result = command.toString()
-            assertThat(result, containsString("tfplan-dev"))
+        @Nested
+        public class WithoutDirectoryProvided {
+            @Test
+            void decoratesTheTerraformEnvironmentStage()  {
+                PassPlanFilePlugin plugin = new PassPlanFilePlugin()
+                def environment = spy(new TerraformEnvironmentStage())
+                plugin.apply(environment)
+
+                verify(environment, times(1)).decorate(eq(TerraformEnvironmentStage.PLAN_COMMAND), any(Closure.class))
+                verify(environment, times(1)).decorate(eq(TerraformEnvironmentStage.APPLY_COMMAND), any(Closure.class))
+            }
+
+            @Test
+            void addsArgumentToTerraformPlan() {
+                PassPlanFilePlugin plugin = new PassPlanFilePlugin()
+                TerraformPlanCommand command = new TerraformPlanCommand("dev")
+                plugin.apply(command)
+
+                String result = command.toString()
+                assertThat(result, containsString("-out=tfplan-dev"))
+            }
+
+            @Test
+            void addsArgumentToTerraformApply() {
+                PassPlanFilePlugin plugin = new PassPlanFilePlugin()
+                TerraformApplyCommand command = new TerraformApplyCommand("dev")
+                plugin.apply(command)
+
+                String result = command.toString()
+                assertThat(result, containsString("tfplan-dev"))
+            }
```

Comment:
`apply` hasn't changed at all with this PR, just `stash`/`unstash`.  It feels like it would make more sense to apply the tests to those contexts, and leave `apply` as-is. 

In the case of testing `stash`/`unstash`, you could validate that by default, `MockWorkflow.dir` is given './', and then when you call `PassPlanFilePlugin.withDirectory('blah')`, `MockWorkflow.dir` is given 'blah'